src/components/network-chart/NetworkChartRow.js

```diff
@@ -220,33 +192,107 @@ class NetworkChartRow extends React.PureComponent<NetworkChartRowProps, State> {
 
     switch (fileExt) {
       case '.js':
-        return 'networkChartRowItemJs';
+        return 'Js';
       case '.css':
-        return 'networkChartRowItemCss';
+        return 'Css';
       case '.gif':
       case '.png':
       case '.jpg':
       case '.jpeg':
       case '.svg':
-        return 'networkChartRowItemImg';
+        return 'Img';
       case '.html':
-        return 'networkChartRowItemHtml';
+        return 'Html';
       default:
-        return '';
+        return 'Other';
     }
   }
 
+  _createMarkerMetaData(
+    marker: TracingMarker
+  ): {
+    requestQueue: number,
+    request: number,
+    response: number,
+    type: string,
+  } {
+    //change object to actuall definition
+    const { networkPayload } = this.props;
+
+    if (!networkPayload) {
+      return {
+        requestQueue: 0,
+        request: 0,
+        response: 0,
+        type: '',
+      };
+    }
+
+    const start = networkPayload.startTime;
+
+    // A marker does not always contain the same set of networkPayload on the start of
+    // the connection.
+    const queueStart =
+      networkPayload.secureConnectionStart ||
+      networkPayload.tcpConnectEnd ||
+      networkPayload.connectStart ||
+      networkPayload.domainLookupStart ||
+      0;
+
+    // The default for width of the phases is always zero.
+    let requestQueue: number = 0;
+    let request: number = 0;
+    let response: number = 0;
+
+    // Timestamps from network markers aren't adjusted when processing
+    // the gecko profile format. This is the reason why the start timestamp of a
+    // marker can be after the start timestamp of the request.
+    // To prevent false networkPayload visualization, we need this workaround.
+    // See https://github.com/devtools-html/perf.html/issues/1493 for more detail.
+    if (
+      networkPayload.requestStart &&
+      start < networkPayload.requestStart &&
+      networkPayload.responseStart &&
+      start < networkPayload.responseStart
+    ) {
+      request = networkPayload.responseStart - networkPayload.requestStart; // create object to throw into tooltip
+    }
+
+    if (
+      networkPayload.responseEnd &&
+      start < networkPayload.responseEnd &&
+      networkPayload.responseStart &&
+      start < networkPayload.responseStart
+    ) {
+      response = networkPayload.responseEnd - networkPayload.responseStart;
+    }
+
+    if (queueStart && start < queueStart && queueStart > 0) {
+      requestQueue = queueStart - start;
+    }
+
+    const markerMeta = {
+      requestQueue: requestQueue,
+      request: request,
+      response: response,
+      type: this._identifyType(marker.name),
+    };
+
+    return markerMeta;
+  }
+
   render() {
     const { index, marker, markerStyle, networkPayload } = this.props;
-
     const evenOddClassName = index % 2 === 0 ? 'even' : 'odd';
+    const markerMetaData = this._createMarkerMetaData(marker);
```

Comment:
This method creates a new object every time it is called. This will force the `<NetworkChartRowBar>` component to re-render every time it is called, which can quickly turn into a performance problem. This method should be memoized.

I believe this blog post has some more information about this issue, and the idea of memoizing derived data: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization

e.g.

```diff
diff --git a/src/components/network-chart/NetworkChartRow.js b/src/components/network-chart/NetworkChartRow.js
index b3dc7dbb..1382abcc 100644
--- a/src/components/network-chart/NetworkChartRow.js
+++ b/src/components/network-chart/NetworkChartRow.js
@@ -6,6 +6,7 @@
 import * as React from 'react';
 import MarkerTooltipContents from '../shared/MarkerTooltipContents';
 import Tooltip from '../shared/Tooltip';
+import memoize from 'memoize-immutable';
 
 import type { CssPixels } from '../../types/units';
 import type { ThreadIndex } from '../../types/profile';
@@ -208,14 +209,12 @@ class NetworkChartRow extends React.PureComponent<NetworkChartRowProps, State> {
     }
   }
 
-  _createMarkerMetaData(
-    marker: TracingMarker
-  ): {
+  _createMarkerMetaData = memoize((marker: TracingMarker): {
     requestQueue: number,
     request: number,
     response: number,
     type: string,
-  } {
+  } => {
     //change object to actuall definition
     const { networkPayload } = this.props;
 
@@ -279,7 +278,7 @@ class NetworkChartRow extends React.PureComponent<NetworkChartRowProps, State> {
     };
 
     return markerMeta;
-  }
+  });
 
   render() {
     const { index, marker, markerStyle, networkPayload } = this.props;
```